lib/datasets/nagoya-university-conversation-corpus.rb

```diff
@@ -0,0 +1,108 @@
+require_relative 'dataset'
+require_relative 'zip-extractor'
+
+module Datasets
+  class NagoyaUniversityConversationCorpus < Dataset
+    Data = Struct.new(
+      :name,
+      :date,
+      :place,
+      :participants,
+      :relationships,
+      :note,
+      :sentences
+    )
+
+    Participant = Struct.new(
+      :id,
+      :attribute,
+      :birthplace,
+      :residence
+    )
+
+    Sentence = Struct.new(
+      :participant_id,
+      :content
+    )
+
+    def initialize
+      super()
+      @metadata.id = 'nagoya-university-conversation-curpus'
+      @metadata.name = 'Nagoya University Conversation Curpus'
+      @metadata.url = 'https://mmsrv.ninjal.ac.jp/nucc/'
+      @metadata.licenses = ['CC-BY-NC-ND-4.0']
+      @metadata.description = <<~DESCRIPTION
+        The "Nagoya University Conversation Corpus" is a corpus of 129 conversations,
+        total about 100 hours of chatting among native speakers of Japanese,
+        which is converted into text.
+      DESCRIPTION
+    end
+
+    def each
+      return to_enum(__method__) unless block_given?
+
+      open_data do |text_file|
+        yield(parse_file(text_file))
+      end
+    end
+
+    private
+
+    def open_data
+      data_path = cache_dir_path + 'nucc.zip'
+      data_url = 'https://mmsrv.ninjal.ac.jp/nucc/nucc.zip'
+      download(data_path, data_url)
+      zip_file = Zip::File.open(data_path)
+      zip_file.each do |entry|
+        yield(entry) if entry.file?
+      end
+    end
+
+    def parse_file(text_file)
+      data = Data.new
+      participants = []
+      sentences = []
+
+      text_file.get_input_stream.each do |input|
+        input.each_line(chomp: true) do |line|
+          line = line.force_encoding('utf-8')
+          if line.include?('＠データ')
```

Comment:
Can we use `start_with?` instead of `include?` here?

Reply:
I did not know this method, and we can use this. Thank you.